test/Mono.Linker.Tests.Cases/DataFlow/CompilerGeneratedCodeInPreservedAssemblyWithWarning.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	// Similar to CompilerGeneratedCodeInPreservedAssembly, but with warnings
+	// produced while marking the compiler generated code.
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+	[SetupLinkerArgument ("--enable-opt", "ipconstprop")]
+	[SetupLinkerDescriptorFile ("CompilerGeneratedCodeInPreservedAssembly.xml")]
+	class CompilerGeneratedCodeInPreservedAssemblyWithWarning
+	{
+		[ExpectedWarning ("IL2026", "--" + nameof (Inner) + "." + nameof (Inner.WithLocalFunctionInner) + "--")]
+		[ExpectedWarning ("IL2026", "--" + nameof (WithLocalFunction) + "--")]
+		public static void Main ()
+		{
+			Inner.WithLocalFunctionInner ();
+			WithLocalFunction ();
+		}
+
+		class Inner
+		{
+			// In this case the compiler generated state will see the modified body,
+			// and will not associate the local function with the user method.
+			// Generic argument warnings from the local function will not be suppressed
+			// by RUC on the user method.
+			[RequiresUnreferencedCode ("--" + nameof (Inner) + "." + nameof (WithLocalFunctionInner) + "--")]
+			public static void WithLocalFunctionInner ()
+			{
+				if (AlwaysFalse) {
+					LocalWithWarning<int> ();
+				}
+
+				[ExpectedWarning ("IL2091", ProducedBy = ProducedBy.Trimmer)]
```

Comment:
We should probably create an issue for this and link it here.

BTW: I don't think it's important to fix this... bust still.